resources/page: Add :slugorfilename attribute

dawidpotocki opened the pull request:
Fixes #4739, #385

It adds a new `:slugelsefile` attribute that behaves the same way as `:slug`
with the only difference being that it fallbacks to filename instead of title.

Both issues are closed because they got marked as stale… but I don't see any
point in opening an another issue.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/gohugoio/hugo?pullRequest=10003) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/gohugoio/hugo?pullRequest=10003) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/gohugoio/hugo?pullRequest=10003) it.</sub>

bep commented:
Yes, this is reasonable, but:

1. I suggest we name it `:slugorfilename`
2. We need a simple test case for it (there should be a test that can be adapted)

dawidpotocki commented:
1. Done.
2. I now added tests to check if `:slugorfilename` extends to `slug`
when set and also added tests for checking if `:slugorfilename` and
`:slug` properly fallback to filename and title respectively.

bep commented:
This looks good, thanks.
